src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/LatestCalculationService.kt

```diff
@@ -100,6 +100,20 @@ class LatestCalculationService(
     sentenceAndOffences: List<SentenceAndOffences>?,
     breakdown: CalculationBreakdown?,
   ): LatestCalculation {
+    val dates = releaseDates(prisonerCalculation)
+    return LatestCalculation(
+      prisonerId,
+      bookingId,
+      prisonerCalculation.calculatedAt,
+      calculationReference,
+      location,
+      reason,
+      calculationSource,
+      calculationResultEnrichmentService.addDetailToCalculationDates(dates, sentenceAndOffences, breakdown).values.toList(),
+    )
+  }
+
+  fun releaseDates(prisonerCalculation: OffenderKeyDates): List<ReleaseDate> {
```

Comment:
LatestestCalculationService seems like it's badly named now seeing as these aren't the latest :)
Maybe extract this logic to a different/new service or rename this and the methods to reflect new behaviour

Reply:
Created a new service - [OffenderKeyDatesService.kt] and moved the common logic of creating the released dates. A simple refactor was done on LatestCalculationService to use the common logic 
[releaseDates, createASLEDIfWeCan].